### ISSUE TITLE
Ansible Inventory Runs as Rundeck User Instead of Specified User

### ISSUE THREAD

**Issue by vexter0944** [rundeck/rundeck#5711]
**Describe the bug**
When you install rundeck and setup the 'Ansible Resource Model Source' to create the nodes for a project, it runs the playbook out of the /tmp folder and does so as the 'rundeck' user, not as the user specified in the 'Ansible Resource Model Source' ssh setup.  Aka - the playbook runs the inventory gather using the hosts file from ansible as 'rundeck' not as 'myuser'.  

As a result, the ssh keys are not in the known_hosts file under the 'rundeck' user and so the ssh fails using the ssh key from var/lib/rundeck/.ssh/id_rsa that was created using 'myuser' with an error like this:
"msg": "Failed to connect to the host via ssh: Host key verification failed.\r\n","unreachable": true

If you add 
[ssh_connection]
ssh_args = -o StrictHostKeyChecking=accept-new
to the ansible.cfg - then the 'rundeck' user will auto accept new keys and everything works as expected.  

**My Rundeck detail**

* Rundeck version: 3.2.1-20200113
* install type: Installed using apt-get from here - https://docs.rundeck.com/docs/administration/install/linux-deb.html#open-source-rundeck
* OS Name/version: Ubuntu 18.04 Server LTS
* DB Type/version: Built-In -dataSource.url: jdbc:h2:file:/var/lib/rundeck/data/rundeckdb;MVCC=true

**To Reproduce**
1. Install Ubuntu Server (18.04.3. LTS) with user 'myuser' during the install when asked
2. Install Rundeck Using documentation from: https://docs.rundeck.com/docs/administration/install/linux-deb.html#open-source-rundeck
3. Install Ansible using documentation from: https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html
4. Edit /etc/ansible/hosts file and add in a host for it to use/pickup (fqdn or ip - no difference)
5.  Setup the sshkeys under 'myuser' account on rundeck server: similar to this: https://www.digitalocean.com/community/tutorials/how-to-set-up-ssh-keys-on-ubuntu-1804
6.  Copy out ssh keys via ssh-copy-id (logged in as 'myuser@rundeckserver.fqdn.com') 'myuser'@client01.fqdn.com and test ssh from rundeck server after doing ssh-copy-id to client01.fqdn.com
7.  Copy /home/myuser/.ssh/id_rsa to /var/lib/rundeck/.ssh/id_rsa (have to use sudo -i on rundeck server since I have no rights as 'myuser' to that folder)
8.  Setup 'Ansible Resource Model Source' and tell it to use 'myuser' and the /var/lib/rundeck/.ssh/id_rsa as privatekey setup and /etc/ansible/hosts for inventory
9.  It 'should' show in a tail -f /var/log/rundeck/service.log at this point when you save the 'Ansible Resource Model Source' setup the failures of the ssh key as described above.
10.  Now go back and add the following to the ansible.cfg file and I rebooted, then it should pull in the nodes when the job/setup refreshed for the 'Ansible Resource Model Source'.
[ssh_connection]
ssh_args = -o StrictHostKeyChecking=accept-new

**Expected behavior**
I expected the nodes to be picked up from the  'Ansible Resource Model Source'  using 'myuser@host' ssh keys I had setup while logged into the rundeck server as 'myuser' and when I did that ssh-copy-id I added each host to the known_hosts under 'myuser' and so the fingerprint is accepted when the ssh connection is initiated. 

**Screenshots**
None at this time

**Desktop (please complete the following information):**
 - OS: Win 10 Pro
 - Browser Chrome
 - Version 79.0.3945.88 (Official Build) (64-bit)

**Additional context**
I tested this by running chsh -s /bin/bash rundeck (as root) and then sudo su - rundeck
which allowed me to get a shell as the rundeck user.  I noted there was no known_hosts in the ~/.ssh/ folder for the rundeck user.  I then initiated the ssh 'myuser@client' command which asked me to accept the fingerprint.  When I did accept the fingerprint - then re-initiated the 'Ansible Resource Model Source'  node import - the node I had not accepted the fingerprint from failed, the 2nd node I had accepted the fingerprint from worked.  So the known_hosts file that is being used is not 'myuser' but the 'rundeck' user and so it explained the errors I was getting from the ansible inventory playbook because it couldn't make a ssh connection since the client was not in the known_hosts for the 'rundeck' user only in the known_hosts for 'myuser'

**Comment by MegaDrive68k** [comment]
Hi @vexter0944 

Did you try defining ansible_user in your ansible hosts file? (for each remote node) take a look at this:

https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html#id10

In case that you're the same account on all servers, you can define one user for all remote nodes with remote_user=test at [defaults] section (ansible.cfg)

https://stackoverflow.com/a/34336459/10426011

Both options work on my test environment.

Kind regards!

**Comment by vexter0944** [comment]
Sort of - I was not aware of the ansible_user in the ansible hosts file - I can try that when I get a minute and report back.  

As to the remote_user=test at [defaults] section (ansible.cfg) - I tried it as remote_user=test but that is how it looked in my ansible.cfg - I did not have test@host or anything after the test so it looked like:
[defaults]
remote_user=test

It did not help when I tried it.

Thanks for responding.  I'm willing to get logs or something if that helps with this or even a screenshare if you want to work together in real time for a bit and see my environment in action.

Thanks so much for all the help and input!
